tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/ProjectEncodingTest.java

```diff
@@ -0,0 +1,238 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Ingo Mohr - Issue #166 - Add Preference to Turn Off Warning-Check for Project Specific Encoding
+ *******************************************************************************/
+package org.eclipse.core.tests.resources;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import org.eclipse.core.internal.resources.PreferenceInitializer;
+import org.eclipse.core.internal.resources.ValidateProjectEncoding;
+import org.eclipse.core.internal.utils.Messages;
+import org.eclipse.core.resources.*;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+import org.eclipse.osgi.util.NLS;
+import org.hamcrest.*;
+import org.junit.Test;
+
+/**
+ * Test for integration of marker
+ * {@link org.eclipse.core.resources.ResourcesPlugin#PREF_MISSING_ENCODING_MARKER_SEVERITY}.
+ */
+public class ProjectEncodingTest extends ResourceTest {
+
+	private static final int IGNORE = -1;
+
+	private IProject project;
+
+	@Override
+	protected void tearDown() throws Exception {
+		if (project != null) {
+			project.delete(true, null);
+		}
+		InstanceScope.INSTANCE.getNode(ResourcesPlugin.PI_RESOURCES).putInt(
+				ResourcesPlugin.PREF_MISSING_ENCODING_MARKER_SEVERITY,
+				PreferenceInitializer.PREF_MISSING_ENCODING_MARKER_SEVERITY_DEFAULT);
+		InstanceScope.INSTANCE.getNode(ResourcesPlugin.PI_RESOURCES).flush();
+		getWorkspace().getRoot().refreshLocal(IResource.DEPTH_INFINITE, null);
+		ensureDoesNotExistInWorkspace(getWorkspace().getRoot());
```

Comment:
I didn't write that method nor change any line of code in that class.

From its impl: if the resource exists -> delete it.

=> I'm with you about this method signature not being clear enough. But changing that class is entirely out of scope of my PR. And sure: Modifying some other class might not hurt. But I'm already working on the 16th (yes, 16th) Pull Request build for this feature which is in parts because 4 reviewers had different opions about things (I still appreciate every feedback, though - just to be clear), and I certainly want to extend the discussion to code that I didn't modify myself.

Reply:
> I didn't write that method nor change any line of code in that class.
> 
> From its impl: if the resource exists -> delete it.

You mean you copied code you don't understand. All good, I don't understand it either. To me it reads like "delete the workspace", which seems odd, but maybe I misunderstand. I see a lot of other tests do the same, so I assume it doesn't hurt.

Reply:
> You mean you copied code you don't understand

Not at all. _You are referring to a method that is not in my PR but in a class that existed before_. And when I look into that method I understand what it does - and I decided that if (apparently so) that method signature was good enough for other people's calls, then it's good enough for my call, as well.

Reply:
> To me it reads like "delete the workspace", which seems odd, but maybe I misunderstand

I'm fully w/ you on that one, though. :)

Reply:
I think I understood you wrong. I first thought your feedback was on the method signature. But perhaps it was more on "why call that method at all".

I deleted that line now. I took it over from ``NatureTest`` (which was initially suggested as template for by change), but we don't need that call at all.